lib/build/bundler.js

```diff
@@ -232,7 +232,12 @@ exports.Bundler = class {
   }
 
   write() {
-    return Promise.all(this.bundles.map(x => x.write(this.project.build.targets[0])));
+    return Promise.all(this.bundles.map(bundle => bundle.write(this.project.build.targets[0])))
```

Comment:
It was actually like this except that the write was asynchronous. Once the vendor bundle was reached it wrote the revisions to the html but the vendor bundle doesn't know about the revisions of other files. I don't wanted to introduce code to pass revisions of other bundles into the last bundle because it felt a bit dirty and therefore I delayed the write process. 

Regarding the sync write: I started with that solution but it were a bit hard because you use a mkdir function in the async write which is based on callbacks. I don't wanted to create a sync write function without this mkdir stuff as it would mean different behavior for those functions.

Reply:
I see.

I have some concern about current bundle writes.
All bundles writes are in parallel (`Promise.all`), they are fine because they deal with different files.
But now they all wants to touch same file, if:
```
bundle A reads index file
bundle B reads index file
bundle A writes modified index file (if A writes after B reads, this modification will be lost)
bundle B writes modified index file
```

I think we need to ensure the operation on index file is in serial (chain of promises), not parallel.
Serialise all bundle writes may be an overkill. But serialise index file modifications is needed, maybe extract it out and do it after all bundles writes (I mean the bundle files themselves) finished?

Reply:
For efficiency, maybe move the logic to bundler? So you can do just one write for all bundles, instead of rewrite index file multiple times.